app/queries/with_one_matching_value_for_each_key.rb

```diff
@@ -0,0 +1,45 @@
+class WithOneMatchingValueForEachKey
+  def initialize(query_field: :tags, scope: SubscriberList)
+    raise ArgumentError.new("query_field must be `:tags` or `:links`") unless %i{tags links}.include?(query_field)
+
+    @query_field = query_field.to_sym
+    @scope = scope
+  end
+
+  # Find all lists in which at least one match in the supplied list.
```

Comment:
`in which at least one value in query_field matches a value in the supplied hash`